src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -41,6 +42,25 @@ public class RandomAttribute : DataAttribute, IParameterDataSource
     {
         private RandomDataSource _source;
         private int _count;
+        private bool _distinct;
+
+        /// <summary>
+        /// If true, the random values will be all different
+        /// </summary>
+        public bool Distinct
+        {
+            get
+            {
+                return _distinct;
+            }
+            set
+            {
+                _distinct = value;
+
+                if (_source != null)
+                    _source.Distinct = _distinct;
```

Comment:
Is there any reason to set this here since it's always being set just in time in `GetData`? I'd rather take Occam's razor to this if possible.